test/e2e/beat/config_test.go

```diff
@@ -100,6 +102,90 @@ heartbeat.monitors:
 	test.Sequence(nil, test.EmptySteps, esBuilder, hbBuilder).RunSequential(t)
 }
 
+func TestBeatSecureSettings(t *testing.T) {
+	skipUnsupportedVersion(t)
+
+	name := "test-beat-secure-settings"
+
+	esBuilder := elasticsearch.NewBuilder(name).
+		WithESMasterDataNodes(3, elasticsearch.DefaultResources)
+
+	testPodBuilder := beat.NewPodBuilder(name)
+
+	secretName := "secret-agent"
+	agentName := "test-agent-name-xyz"
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: test.Ctx().ManagedNamespace(0),
+		},
+		Data: map[string][]byte{
+			"AGENT_NAME_VAR": []byte(agentName),
+		},
+	}
+
+	fbBuilder := beat.NewBuilder(name, filebeat.Type).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithSecureSettings(secretName).
+		WithESValidations(
+			beat.HasEventFromBeat(filebeat.Type),
+			beat.HasEventFromPod(testPodBuilder.Pod.Name),
+			beat.HasMessageContaining(testPodBuilder.Logged),
+			beat.HasEvent("agent.name:"+agentName),
+		)
+
+	config := `
+name: ${AGENT_NAME_VAR}
+filebeat:
+  autodiscover:
+    providers:
+    - hints:
+        default_config:
+          paths:
+          - /var/log/containers/*${data.kubernetes.container.id}.log
+          type: container
+        enabled: true
+      host: ${NODE_NAME}
+      type: kubernetes
+processors:
+- add_cloud_metadata: {}
+- add_host_metadata: {}
+`
+
+	fbBuilder = applyConfigYaml(t, fbBuilder, config)
+
+	beforeFn := func(k *test.K8sClient) test.StepList {
+		return test.StepList{
+			{
+				Name: "Create secure settings secret",
+				Test: func(t *testing.T) {
+					_ = k.Client.Delete(secret)
+					require.NoError(t, k.Client.Create(secret))
+				},
+			},
+			// Keystore should be empty
+			test.CheckKeystoreEntries(
```

Comment:
This `Step` is called before any Pod is created. There is not keystore entries because there is no Pod. I think this is not what we we want to test ?